_pytest/runner.py

```diff
@@ -176,24 +176,27 @@ def check_interactive_exception(call, report):
 def call_runtest_hook(item, when, **kwds):
     hookname = "pytest_runtest_" + when
     ihook = getattr(item.ihook, hookname)
-    return CallInfo(lambda: ihook(item=item, **kwds), when=when)
+    return CallInfo(lambda: ihook(item=item, **kwds), when=when, usepdb=item.config.getvalue("usepdb"))
 
 
 class CallInfo(object):
     """ Result/Exception info a function invocation. """
     #: None or ExceptionInfo object.
     excinfo = None
 
-    def __init__(self, func, when):
+    def __init__(self, func, when, usepdb=None):
```

Comment:
What about passing in the whole `item`?
Would make it more flexible for future changes.

Reply:
I don't mind changing it.

I figured it was cleaner to pass the minimum information I need to accomplish the task, passing more seemed like premature optimization.

Reply:
I personally prefer to pass just what you need as well. I would name the parameter differently as well, because it is more about treating `KeyboardInterrupt` as a special case... how about `stop_on_keyboard_interrupt=True`?

Reply:
Sounds good, but in either case we are going to stop on keyboard interrupt. Either we capture the exception or we re-raise it. How about `treat_keyboard_interrupt_as_exception`?